UM-USElab/gradecraft-development
# Group Submissions Exports

### Status
**NEEDS REVIEW**

### Description
This PR adds the ability to generate submissions exports for group-submitted assignments. Students have been decoupled from the submissions export generation workflow and are now referred to "submitters".

Additional logic has been added to the Group and User models to clarify the roles each plays in the guise of a submitter. Relevant queries in Assignment for fetching appropriate resources have also been merged to reduce duplication.

### Deploy Notes
Migrations must be run on this branch prior to deployment.

### Migrations
YES

### Steps to Test or Reproduce
1. Visit an assignment that has group submissions.
2. Click the 'Options' button, then 'Download Group Submissions'.
3. Visit 'Course Data Exports' on the left-hand navigation menu.
4. A new submissions export for the given assignment should appear. The user requesting export will receive an email within :30 seconds or so of request depending on how many files are attached. Refreshing the page will offer a download link for the completed export.

### Impacted Areas in Application
* Group assignments.
